util/PaginationEmbed.js

```diff
@@ -0,0 +1,564 @@
+   */
+  async build() {
+    this
+      .setAuthorisedUser(this.authorisedUser)
+      .setChannel(this.channel)
+      .setClientMessage(this.clientMessage.message, this.clientMessage.content)
+      .setArray(this.array)
+      .setElementsPerPage(this.elementsPerPage)
+      .showPageIndicator(this.pageIndicator)
+      .setTimeout(this.timeout);
+
+    this.pages = Math.ceil(this.array.length / this.elementsPerPage);
+    this.setPage(this.page);
+
+    if (!(this.page >= 1 && this.page <= this.pages)) throw new Error("Invalid page.");
+
+    const message = this.clientMessage.message
+      ? await this.clientMessage.message.edit(this.clientMessage.content)
+      : await this.channel.send(this.clientMessage.content);
+    this.setClientMessage(message, this.clientMessage.content);
+
+    const permissions = ["ADD_REACTIONS", "MANAGE_MESSAGES", "EMBED_LINKS"];
+    const missing = message.channel.permissionsFor(message.client.user).missing(permissions);
+
+    if (missing.length)
+      throw new Error(`Cannot invoke Pagination class without required permissions: ${missing.join(", ")}`);
+
+    const isValidFields = Array.isArray(this.fields) && Boolean(this.fields.length);
+
+    if (!isValidFields) throw new Error("Cannot invoke Pagination class without initialising at least one field.");
+
+    const fields = this.fields;
+    this.fields = [];
+
+    for (let i = 0; i < fields.length; i++) {
+      const field = fields[i];
+
+      if (typeof field.value === "function")
+        this.formatField(field.name, field.value, field.inline);
+      else
+        this.addField(field.name, field.value, field.inline);
+    }
+
+    await this._loadList();
+  }
+
+  /**
+   * Adds a field to the embed.
+   * Same as MessageEmbed.addField, but value takes a function instead.
+   * @param {string} name - Name of the field.
+   * @param {Function} value - Value of the field. Function for Array.prototype.map().join("\n").
+   * @param {boolean} [inline=true] - Whether the field is inline with other field or not.
+   * @default true - Default for inline
+   * @returns {PaginationEmbed} - Instance of PaginationEmbed
+   */
+  formatField(name, value, inline = true) {
+    if (typeof value !== "function") console.warn("Warning: formatField() only accepts function type for field value.");
+
+    this.fields.push({ name, value, inline });
+
+    return this;
+  }
+
+  /**
+   * Sets the array of elements to paginate.
+   * @param {Array} array - An array of elements to paginate.
+   * @returns {PaginationEmbed} - Instance of PaginationEmbed
+   */
+  setArray(array) {
+    const isValidArray = Array.isArray(array) && Boolean(array.length);
+
+    if (!isValidArray) throw new Error("Cannot invoke Pagination class without initialising the array to paginate.");
+
+    this.array = array;
+
+    return this;
+  }
+
+  /**
+   * Set the authorised person to navigate the pages.
+   * @param {Object} [user=null] - The user object.
+   * @default null
```

Comment:
This is unnecesary. `@param {User} [user=null]` already states it defaults to `null`. Same to the occurrences below